cg/meta/workflow/rnafusion.py

```diff
@@ -0,0 +1,315 @@
+        stub: bool,
+        input: str,
+        outdir: str,
+        genomes_base: str,
+        trim: bool,
+        fusioninspector_filter: bool,
+        all: bool,
+        pizzly: bool,
+        squid: bool,
+        starfusion: bool,
+        fusioncatcher: bool,
+        arriba: bool,
+        dry_run: bool = False,
+    ) -> None:
+        """Execute RNAFUSION run analysis with given options"""
+
+        options = self.__build_command_str(
+            self.get_verified_arguments(
+                case_id=case_id,
+                work_dir=work_dir,
+                resume=resume,
+                profile=profile,
+                with_tower=with_tower,
+                stub=stub,
+                input=input,
+                outdir=outdir,
+                genomes_base=genomes_base,
+                trim=trim,
+                fusioninspector_filter=fusioninspector_filter,
+                all=all,
+                pizzly=pizzly,
+                squid=squid,
+                starfusion=starfusion,
+                fusioncatcher=fusioncatcher,
+                arriba=arriba,
+            )
+        )
+        nextflow_options = self.__build_command_str(
+            self.get_verified_arguments_nextflow(
+                case_id=case_id,
+                log=log,
+            )
+        )
+        command = ["run", self.nfcore_pipeline_path]
+        parameters = (
+            nextflow_options
+            + ["-bg", "-q"]
+            + command
+            + options
+            + [
+                " > "
+                + str(self.get_case_path(case_id))
+                + "/"
+                + case_id
+                + "-stdout.log 2> "
+                + str(self.get_case_path(case_id))
+                + "/"
+                + case_id
+                + "-stdout.err  < /dev/null & "
+            ]
+        )
+        self.process.run_command(parameters=parameters, dry_run=dry_run)
+
+    def get_deliverables_file_path(self, case_id: str) -> Path:
+        """Returns a path where the rnafusion deliverables file for the case_id should be located.
+
+        (Optional) Checks if deliverables file exists
+        """
+        return Path(self.get_case_path(case_id), case_id + "_deliverables.yaml")
+
+    def get_template_deliverables_file(self, rnafusion_bundle_template: Path) -> pd.DataFrame:
+        """Read deliverables file template and output to dataframe format"""
+        return pd.read_csv(rnafusion_bundle_template)
+
+    def edit_template_deliverables_file(
+        self, case_id: str, deliverables_template: pd.DataFrame
+    ) -> pd.DataFrame:
+        """Replace PATHTOCASE and CASEID from template deliverables file to corresponding strings, add path_index column"""
+        edited_deliverables = deliverables_template.replace(
+            {"PATHTOCASE": str(self.get_case_path(case_id))}, regex=True
+        )
+        edited_deliverables = edited_deliverables.replace({"CASEID": case_id}, regex=True)
```

Comment:
You can make this into a loop over a dict instead. Then you only add new key-value pairs instad of new lines iif you have to replace something else in the future. Something like this pseudo code:
```
replace_map: dict[str, str] = {"PATHTOCASE": str(self.get_case_path(case_id))},
"CASEID": case_id}

for str_to_replace, with_value in replace_map.item():
    edited_deliverables: pd.DataFrame = edited_deliverables(str_to_replace, with_value, , regex=True)
```

Reply:
[a9bcdaa](https://github.com/Clinical-Genomics/cg/pull/1541/commits/a9bcdaa809d33b7850d0498921896f84305c3b36)